polaris/polaris/utils.py

```diff
@@ -185,95 +148,39 @@ def create_stellar_deposit(transaction: Transaction) -> bool:
             return False
     # otherwise, create the envelope and sign it with the distribution account's secret
     else:
-        multisig = False
         distribution_acc, _ = get_account_obj(
             Keypair.from_public_key(transaction.asset.distribution_account)
         )
         envelope = create_transaction_envelope(transaction, distribution_acc)
         envelope.sign(transaction.asset.distribution_seed)
-        transaction.envelope_xdr = envelope.to_xdr()
 
     try:
-        return submit_stellar_deposit(transaction, multisig=multisig)
-    except RuntimeError as e:
-        transaction.status_message = str(e)
+        submit_stellar_deposit(transaction, envelope)
```

Comment:
```suggestion
        submit_stellar_deposit(transaction, envelope)
        return True
```

Reply:
Whats your reasoning?

Reply:
Yea I can do that. Generally I like to only put the code that could possibly raise an exception in the `try`.

Reply:
It's not a big deal now actually since its A) reachable code, and B) I understand how and why this is reachable.
Also I agree with your isolation philosophy.